File: app/src/main/java/com/manrique/trailerstock/model/VentaDetalle.java
package com.manrique.trailerstock.model;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(
        tableName = "venta_detalles",
        foreignKeys = @ForeignKey(
                entity = Venta.class,
                parentColumns = "id",
                childColumns = "ventaId",
                onDelete = ForeignKey.CASCADE
        )
)
public class VentaDetalle {
    @PrimaryKey(autoGenerate = true)
    private int id;

    private int ventaId;
    private int productoId;
    private int cantidad;
    private double precioUnitario; // El precio al que se vendió ese día

    public VentaDetalle(int ventaId, int productoId, int cantidad, double precioUnitario) {
        this.ventaId = ventaId;
        this.productoId = productoId;
        this.cantidad = cantidad;
        this.precioUnitario = precioUnitario;
    }

    // Getters y Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getVentaId() { return ventaId; }
    public void setVentaId(int ventaId) { this.ventaId = ventaId; }
    public int getProductoId() { return productoId; }
    public void setProductoId(int productoId) { this.productoId = productoId; }
    public int getCantidad() { return cantidad; }
    public void setCantidad(int cantidad) { this.cantidad = cantidad; }
    public double getPrecioUnitario() { return precioUnitario; }
    public void setPrecioUnitario(double precioUnitario) { this.precioUnitario = precioUnitario; }
}

File: app/src/main/java/com/manrique/trailerstock/data/local/entities/VentaDetalle.kt
package com.manrique.trailerstock.data.local.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

/**
 * Entity que representa el detalle de una venta (productos vendidos).
 * Relación muchos-a-uno con Venta y Producto.
 */
@Entity(
    tableName = "venta_detalles",
    foreignKeys = [
        ForeignKey(
            entity = Venta::class,
            parentColumns = ["id"],
            childColumns = ["venta_id"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Producto::class,
            parentColumns = ["id"],
            childColumns = ["producto_id"]
        )
    ]
)
data class VentaDetalle(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    
    @ColumnInfo(name = "venta_id", index = true)
    val ventaId: Int,
    
    @ColumnInfo(name = "producto_id", index = true)
    val productoId: Int,
    
    @ColumnInfo(name = "cantidad")
    val cantidad: Int,
    
    @ColumnInfo(name = "precio_unitario")
    val precioUnitario: Double,
    
    @ColumnInfo(name = "subtotal")
    val subtotal: Double
) {
    companion object {
        /**
         * Factory method para crear VentaDetalle con cálculo automático del subtotal
         */
        fun create(
            ventaId: Int,
            productoId: Int,
            cantidad: Int,
            precioUnitario: Double
        ): VentaDetalle {
            return VentaDetalle(
                ventaId = ventaId,
                productoId = productoId,
                cantidad = cantidad,
                precioUnitario = precioUnitario,
                subtotal = cantidad * precioUnitario
            )
        }
    }
}
